### PR TITLE
Replaced ListView with RecyclerView.

### LINKED ISSUE #165: Replace ListView with RecyclerView.
The app is still using ListView.
We should use RecyclerView instead. RecyclerView will give us better control over list items and add animations in future.

Should I send a PR.

### PR DESCRIPTION
**Pull request by m-murad**
Fixes #165 
Please review.

### CODE REVIEW

**Comment by m-murad** [comment]
Please don't merge yet,
I created new files while editing the previous ones, in the process I forgot to rewrite the documentation of classes.

Will update the changes in a few minutes.

**Comment by m-murad** [comment]
I have updated the PR.
Please review.

File: mobile/src/main/java/org/hawkular/client/android/fragment/TriggersFragment.java [hunk 1]
```diff
@@ -113,8 +107,7 @@ private void setUpRefreshing() {
         contentLayout.setColorSchemeResources(ColorSchemer.getScheme());
     }
 
-    @Override
-    public void onRefresh() {
+    @Override public void onRefresh() {
```
Comment (KeenWarrior) [inline mobile/src/main/java/org/hawkular/client/android/fragment/TriggersFragment.java]: Revert this wrong reformating

**Comment by m-murad** [comment]
Thanks for pointing this out.
Will do it in a minute or two.

**Review by KeenWarrior — APPROVED** [review]

**Comment by m-murad** [comment]
@garg-anuj 
Thanks for merging I will send a PR for search functionality tonight.

**Pull request merged**